lib/product.rb

```diff
@@ -0,0 +1,20 @@
+class Product
+    attr_reader :category, :name , :quantity ,:unit_price
+    def initialize(category , name, unit_price, quantity)
+        @category = category
+        @name = name
+        @quantity = quantity.to_i
+        @unit_price = unit_price
+    end
+    def total_price
+        unit_price * quantity
+    end
+    
+    def is_hoarded?
+        quantity == 0
+    end
+
+    def hoard
+        @quantity = 0
+    end
```

Comment:
in some methods you use `quantity` and in others you use `@quantity` . These are calling the same value, but it is  convention to use the instance variable to call that value instead of the method.